### DISCUSSION TITLE
Running vale against all modules in an assembly.

### DISCUSSION THREAD

**Discussion by kquinn1204** [redhat-documentation/vale-at-red-hat#30] [General]
Is there a way to run vale across all modules in an assembly. 
Show changes and also export changes to a file.

**Comment by michaelryanpeter** [comment]
In the short term, could you use your `bccutil` or `asciibinder` `html` preview files?

There is almost definitely a more sophisticated way of doing this using git hooks, but this might do what you need for the moment.

**Comment by aireilly** [comment]
> Is there a way to run vale across all modules in an assembly. 
> 
> Show changes and also export changes to a file. 

Could be scripted Kevin!

**Comment by themr0c** [comment]
* When you run vale on a file, behind the scene vale converts the file into HTML and validates the HTML content. You can indeed directly validate the HTML file. But then the report is pointing to lines in the HTML and you need to reconcile it with the source code.

* You can leverage the `atree` output to provide a list of files to pass to vale. See: https://github.com/redhat-documentation/tools/tree/master/atree
